### ISSUE TITLE
Smartparking folder visible in the GitRepository

### ISSUE THREAD

**Issue by vrosu** [PTCInc/thingworx-gitbackup-extension#8]
Repported by @TG-JC
![image](https://user-images.githubusercontent.com/11868471/95329366-7d55b500-08af-11eb-8cc2-b01e35c6ebe8.png)

**Comment by vrosu** [comment]
Confirmed as a bug. The folder is created in parallel to the folder you specify, after you added a repository from the Main Mashup UI, before you even click Pull.

**Comment by vrosu** [comment]
I believe this was a hidden small issue present for a long time.
-In the AddNewRepo service, I'm creating the Thing, doing Enable and Restart, and after that I'm setting the configuration which includes the proper folder.
-in the Java template, I'm using one of the lifecycle methods, "initializeThing", to prepare some things - like initializing the git repository on the disk. That method is executed during the very first enable/restart, before the correct configuration was set, and initializes the repository with the default value of the path - smartparking.

Since this is not really a bug, just a minor annoyance, I'll close this thread with won't fix.
If anybody needs to have this fixed, please reopen this issue in future.
